doc/src/manual/arrays.md

```diff
@@ -879,10 +879,16 @@ slower than multiplication. While some arrays — like [`Array`](@ref) itself 
 are implemented using a linear chunk of memory and directly use a linear index
 in their implementations, other arrays — like [`Diagonal`](@ref) — need the
 full set of cartesian indices to do their lookup (see [`IndexStyle`](@ref) to
-introspect which is which). As such, when iterating over an entire array, it's
-much better to iterate over [`eachindex(A)`](@ref) instead of `1:length(A)`.
-Not only will the former be much faster in cases where `A` is `IndexCartesian`,
-but it will also support [OffsetArrays](https://github.com/JuliaArrays/OffsetArrays.jl), too.
+introspect which is which).
+
+!!! warnings
+
+    When iterating over the entire array with the values of the indices needed, it is
+    better to iterate over [`eachindex(A)`](@ref) instead of `1:length(A)`.
+    Not only will the former be faster in cases where `A` is `IndexCartesian`,
+    but it will also support arrays with custom indexing, such as [OffsetArrays](https://github.com/JuliaArrays/OffsetArrays.jl).
+    If the values of the indices are not needed, on the other hand, it is preferable to iterate over
+    the array's elements (i.e. `for a in A`).
```

Comment:
Can this be shorter and clearer?  "values of the indices" seems a little odd, aren't those just indices... and values the things in the array?

```suggestion
    When iterating over all the indices for an array, it is
    better to iterate over [`eachindex(A)`](@ref) instead of `1:length(A)`.
    Not only will this be faster in cases where `A` is `IndexCartesian`,
    but it will also support arrays with custom indexing, such as [OffsetArrays](https://github.com/JuliaArrays/OffsetArrays.jl).
    If only the values are needed, then is better to just iterate the array directly, i.e. `for a in A`.
```